app/src/main/java/com/quran/labs/androidquran/database/TranslationsDBAdapter.java

```diff
@@ -105,11 +106,31 @@ public void deleteTranslationByFile(String filename) {
   public boolean writeTranslationUpdates(List<TranslationItem> updates) {
     boolean result = true;
     db.beginTransaction();
+
     try {
       for (int i = 0, updatesSize = updates.size(); i < updatesSize; i++) {
         TranslationItem item = updates.get(i);
         if (item.exists()) {
+          int displayOrder = 0;
+
           final Translation translation = item.getTranslation();
+
+          if (item.getDisplayOrder() > -1) {
+            displayOrder = item.getDisplayOrder();
+          } else {
+            // get next highest display order
+            Cursor cursor = db.query(
```

Comment:
Please, consider closing this cursor.

Reply:
oops, yes to both cursor commments

Reply:
you should do this in a try / finally because if an exception is thrown due to the query, we will leak the cursor

Reply:
sorry to nitpick, the try/catch should be before the cursor assignment even - so like:

```java
Cursor cursor = null;
try {
  cursor = ..
} finally {
  DatabasUtils.closeCursor(cursor);
}
```